### DISCUSSION TITLE
Instant App Support

### DISCUSSION THREAD

**Discussion by AndreasBoehm** [fingerprintjs/fingerprintjs-android#52] [Q&A] [answered]
I was trying fingerprintjs and fingerprintjs PRO to with a regular app and an Instant App.
I was not able to detect that a regular app and an Instant App is running on the same device.

Does anyone have experience with that and can help me with that?

**Comment by Alexey-Verkhovsky** [comment, marked as answer]
Hey! I've tested the library on Instant apps, and what I found is that only GSF ID is unavailable. 
So you can use either Android ID or Media DRM ID for getting the ID, it will be the same.

Get them with following code:

```kotlin
val fingerprinter = FingerprinterFactory
		.getInstance(applicationContext, Configuration(version = 3))

fingerprinter.getDeviceId { deviceIdResult ->
          val androidId = deviceIdResult.androidId
          val mediaDrmId = deviceIdResult.mediaDrmId
}
```

You can match instances on one device by IDs, and distinguish regular app from instant app by some additional logic in [instant-only modules](https://developer.android.com/topic/google-play-instant/getting-started/instant-enabled-app-bundle#configure-instant-enabled-modules).  

Hope it helps!

**Reply by AndreasBoehm** [reply]
Hi @Alexey-Verkhovsky ,

thanks for your help!

Your are right about that, i just retried it and it works as you described it.
There must have been a mistake when i compared the IDs for the first time.

I created a PR to fix the infos on the stability table that i did set wrong: https://github.com/fingerprintjs/fingerprintjs-android/pull/53

Will keep some of the ids as unstable because I haven't re-tested device state fingerprint, hardware fingerprint and os fingerprint.
